com.unity.render-pipelines.core/Editor/Volume/VolumeEditor.cs

```diff
@@ -49,42 +78,100 @@ void RefreshEffectListEditor(VolumeProfile asset)
+            using (var scope = new EditorGUI.ChangeCheckScope())
             {
-                EditorGUI.BeginChangeCheck();
-                isGlobal = EditorGUI.Popup(lineRect, label, isGlobal, m_Modes);
-                if (EditorGUI.EndChangeCheck())
-                    m_IsGlobal.boolValue = isGlobal == 0;
+                isGlobal = EditorGUI.Popup(lineRect, Styles.mode, isGlobal, Styles.modes);
+                if (scope.changed)
+                {
+                    if (isGlobal == 0)
+                    {
+                        // Store the used tool to restore the value
+                        m_PreviousTool = Tools.current;
+                        m_IsGlobal.boolValue = true;
+                    }
+                    else
+                    {
+                        // Restore the used tool
+                        Tools.current = m_PreviousTool;
+                        m_IsGlobal.boolValue = false;
+                    }
+                }
             }
-            EditorGUI.EndProperty();
-
+            
             if (isGlobal != 0) // Blend radius is not needed for global volumes
             {
-                if (!actualTarget.TryGetComponent<Collider>(out _))
+                if (actualTarget.TryGetComponent<Collider>(out var collider))
                 {
-                    EditorGUILayout.HelpBox("Add a Collider to this GameObject to set boundaries for the local Volume.", MessageType.Info);
-
-                    if (GUILayout.Button(EditorGUIUtility.TrTextContent("Add Collider"), EditorStyles.miniButton))
-                    {
-                        var menu = new GenericMenu();
-                        menu.AddItem(EditorGUIUtility.TrTextContent("Box"), false, () => Undo.AddComponent<BoxCollider>(actualTarget.gameObject));
-                        menu.AddItem(EditorGUIUtility.TrTextContent("Sphere"), false, () => Undo.AddComponent<SphereCollider>(actualTarget.gameObject));
-                        menu.AddItem(EditorGUIUtility.TrTextContent("Capsule"), false, () => Undo.AddComponent<CapsuleCollider>(actualTarget.gameObject));
-                        menu.AddItem(EditorGUIUtility.TrTextContent("Mesh"), false, () => Undo.AddComponent<MeshCollider>(actualTarget.gameObject));
-                        menu.ShowAsContext();
-                    }
+                    if (!collider.enabled)
+                        CoreEditorUtils.DrawFixMeBox(Styles.enableColliderFixMessage, () => collider.enabled = true);
+                }
+                else
+                {
+                    CoreEditorUtils.DrawFixMeBox(Styles.addColliderFixMessage, AddOverride);
                 }
 
                 EditorGUILayout.PropertyField(m_BlendRadius);
                 m_BlendRadius.floatValue = Mathf.Max(m_BlendRadius.floatValue, 0f);
             }
+            else
+            {
+                // When the volume is global hide all the transform gizmos
+                Tools.current = Tool.None;
```

Comment:
Doing this not only hides transform gizmos, it also prevents their selection in the toolbar - so none of these are selected and clicking them does nothing:

![image](https://user-images.githubusercontent.com/77337829/122340488-fdf2a800-cf4a-11eb-9eb2-2dcf02f2f8ba.png)

For example, now you cannot use the View tool when a Global volume is selected. Is this intended? I'm not sure if "no active tool mode" is a well-defined state. The new overlays UI seems to allow it though (not sure if it was always allowed by the old toolbar UI as well).

Reply:
No we just want the associated gizmo to to be unavailable. It should not be affecting anything else in the world.

Reply:
We do not have a way of disabling the gizmos for transforms, this is the only way, I can hide the first time that they are selected as global, and then if the user clicks this will force to show them. Do you want this @TheoWong-pixel

Reply:
I feel like positioning the volume object is still a valid operation, even if the global volume logic then ignores the position. It can still be useful to position the global volumes in a certain place, e.g. for the purpose of organizing your level how you want it. Otherwise users would have to switch to Local mode, reposition it and then switch back to global. Maybe what Alex is suggesting is a good compromise - hiding the transform handles to indicate the position is irrelevant, but user is not prevented from selecting transform tool and moving the global volume to where they like.

Reply:
Agree with @arttu-peltonen here.
During the time I was working in Game Development we often placed things like global volumes, directional lights, etc in proper places to structure our scenes as we wanted. Not allowing that will introduce a pain for users in my opinion.

Reply:
Wait so the position of a global volume DOES matter? I was under the assumption that it would be the same wherever it is.

Reply:
@TheoWong-pixel I don't think it matters for in terms of its functionality. BUT like @ellioman said, as a gamedev, you want to organize your levels (_spatially_) so you find your stuff and it can be annoying not to be able to move a gameobject, just because it's a global volume that doesn't strictly care about its position.

Reply:
Yes exactly. For the end result Global Volumes, Directional Lights, etc do not matter. But when working in scenes, especially big ones, you want to be able to organize these things just like users want with the hierarchy, project, etc.

Reply:
I will remove this then :)

Reply:
Approving the PR, but I guess this needs to be removed before merging right?